stdlib/public/runtime/SwiftValue.mm

```diff
@@ -363,11 +363,18 @@ - (NSString *)debugDescription {
 - (const Metadata *)_swiftTypeMetadata {
   return getSwiftValueTypeMetadata(self);
 }
-- (NSString *)_swiftTypeName {
+- (id /* NSString */)_swiftTypeName {
   TypeNamePair typeName
     = swift_getTypeName(getSwiftValueTypeMetadata(self), true);
 
-  return [NSString stringWithUTF8String: typeName.data];
+  Class nsStringClass = objc_lookUpClass("NSString");
+  SEL utf8Sel = @selector(stringWithUTF8String:);
+  id (*utf8)(id, SEL, const char*) =
+    (id (*)(id, SEL, const char*))class_getMethodImplementation(
+                                                              nsStringClass,
+                                                              utf8Sel
+                                                              );
```

Comment:
Why go through `class_getMethodImplementation` instead of `objc_msgSend`? And why not use the same `swift_stdlib_boxUTF8CString` trick?

Reply:
I have some vague recollection that I had a reason for this, but I don't think it was a very good reason. I'll see if I can change it to swift_stdlib_boxUTF8CString